### ISSUE TITLE
undefined initial argument in createState results in HOOKSTATE-107

### ISSUE THREAD

**Issue by arslancharyev31** [avkonst/hookstate#125]
Good day.

I have noticed that using `undefined` or `null` as initial value in [createState](https://hookstate.js.org/docs/typedoc-hookstate-core/#createstate) results in [HOOKSTATE-107](https://hookstate.js.org/docs/exceptions/#hookstate-107) exception being thrown. I was wondering if this is intended behaviour or a bug?

Example case:
```typescript
interface Obj {
	str: string
	num: number
}
export const state = createState<Obj | undefined>(undefined);
```

**Comment by arslancharyev31** [comment]
### **Update**

It would seem that the exception is present only when exporting the global state constant. Removing the `export` modifier solves the issues.

**Comment by avkonst** [comment]
`undefined` and `null` are supported values. It should work. Where do you see the exception? What action does cause the exception?

**Comment by arslancharyev31** [comment]
> Where do you see the exception?

It is a full-screen fatal exception, like the ones you see with syntax errors.
> What action does cause the exception?

It is caused when 2 conditions are met: the global state is exported and the initial value is undefined/null

To demonstrate the issue more clearly, I have made a sample CRA project which can be found [here](https://github.com/arslancharyev31/hookstate-107). More specifically, [this line](https://github.com/arslancharyev31/hookstate-107/blob/fa116cdfa3a1a926f8b929005494059c7eb56545/src/state/example.ts#L9) causes the exception.

**Comment by avkonst** [comment]
Fixed in 3.0.4
